### PR TITLE
Fix: GCP IAP Config and Docs

### PR DESCRIPTION
**Pull request by bsaltz**
## Hey, I just made a Pull Request!

This is a fix for #25087. The IAP configuration documentation and `ProxiedSignInPage` doc comment were wrong, but the configuration also needed to be duplicated when using `auth.environment`. According to Patrik's comment on the issue, this is expected for proxy auth providers. I double-checked and confirmed that this now aligns with the AWS ALB provider.

I don't think this is a breaking change since the workaround for the issue creates a forwards-compatible configuration file.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/google/gcp-iap-auth.md [hunk 1]
```diff
@@ -77,6 +77,6 @@ backend.add(import('@backstage/plugin-auth-backend-module-gcp-iap-provider'));
 
 ## Adding the provider to the Backstage frontend
 
-See [Sign-In with Proxy Providers](../index.md#sign-in-with-proxy-providers) for pointers on how to set up the sign-in page, and to also make it work smoothly for local development. You'll use `gcp-iap` as the provider name.
+See [Sign-In with Proxy Providers](../index.md#sign-in-with-proxy-providers) for pointers on how to set up the sign-in page, and to also make it work smoothly for local development. You'll use `gcpiap` as the provider name.
```
Comment (freben) [inline docs/auth/google/gcp-iap-auth.md:80]: These are case sensitive aren't they - should this be `gcpIap`?

File: packages/core-components/src/layout/ProxiedSignInPage/ProxiedSignInPage.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ import { useTranslationRef } from '@backstage/core-plugin-api/alpha';
  */
 export type ProxiedSignInPageProps = SignInPageProps & {
   /**
-   * The provider to use, e.g. "gcp-iap" or "awsalb". This must correspond to
+   * The provider to use, e.g. "gcpiap" or "awsalb". This must correspond to
```
Comment (freben) [inline packages/core-components/src/layout/ProxiedSignInPage/ProxiedSignInPage.tsx:37]: same here

**Review by freben — COMMENTED** [review]
Thanks! Just a casing question

File: docs/auth/google/gcp-iap-auth.md (hunk 1 above)
Comment (bsaltz) [reply docs/auth/google/gcp-iap-auth.md:80]: I used `gcpiap` in my local configuration, and it worked, so I didn't really look any deeper. Looking at `auth-backend/src/providers/router.ts` and `auth-backend/src/service/router.ts`, it's using the default options in `express.Router`, which makes it case-insensitive. I can try it with `gcpIap` to make sure it works. If it does, then I'll update both these docs to make everything consistent.

Comment (bsaltz) [reply docs/auth/google/gcp-iap-auth.md]: It was case-insensitive. Using `gcpIap` worked. I've updated this to `gcpIap` instead of `gcpiap`.

**Review by freben — APPROVED** [review]
Thanks!

**Pull request merged**